integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/ObjectExistsSearchPredicateIT.java

```diff
@@ -210,7 +217,8 @@ public void flattened() {
 				.fetchAllHits();
 
 		// DOCUMENT_2 won't be matched either, since it hasn't any not-null field
-		assertThat( docs ).hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_3 );
+		// DOCUMENT_4 will match, even if the matching field is not a direct field of the targeted path
```

Comment:
`direct field` => `direct child`? Not sure I understood correctly, though.

Same below.

Reply:
yeap. it sounds better